T35078 test_report.py: Generate test report for completed node

JenySadadia opened the pull request:
Fixes https://github.com/kernelci/kernelci-api/issues/130

Depends on https://github.com/kernelci/kernelci-pipeline/pull/66 and https://github.com/kernelci/kernelci-core/pull/1242/

Instead of listening to all node events, listen to the "completed" event.
Display total timeouts to test report.

JenySadadia commented:
Tested on local.

```
kernelci-pipeline-notifier | 06/08/2022 09:04:20 AM UTC [INFO] 2022-06-08 09:04:20.509384  9886142c7a22  Timeout   check-describe
kernelci-pipeline-trigger_completed | 06/08/2022 09:04:20 AM UTC [INFO] {'message': "'Event triggered"}
kernelci-pipeline-notifier | 06/08/2022 09:04:20 AM UTC [INFO] 2022-06-08 09:04:20.522510  9886142c7a22  Pending   checkout
kernelci-pipeline-test_report | 2 0 1
kernelci-pipeline-test_report | 06/08/2022 09:04:20 AM UTC [INFO] Content-Type: multipart/mixed; boundary="===============1230460839464074404=="
kernelci-pipeline-test_report | MIME-Version: 1.0
kernelci-pipeline-test_report | To: kernelci-results-staging@groups.io
kernelci-pipeline-test_report | From: bot@kernelci.org
kernelci-pipeline-test_report | Subject: Kernel CI Test Reports
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | --===============1230460839464074404==
kernelci-pipeline-test_report | Content-Type: text/plain; charset="utf-8"
kernelci-pipeline-test_report | MIME-Version: 1.0
kernelci-pipeline-test_report | Content-Transfer-Encoding: quopri
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | mainline/master v5.19-rc1-11-g9886142c7a22: 2 runs, 0 failures, 1 timeouts
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | test            | result
kernelci-pipeline-test_report | ----------------+-------
kernelci-pipeline-test_report | tarball         | pass
kernelci-pipeline-test_report | check-describe  | timeout
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report |   Tree:     mainline
kernelci-pipeline-test_report |   Branch:   master
kernelci-pipeline-test_report |   Describe: v5.19-rc1-11-g9886142c7a22
kernelci-pipeline-test_report |   URL:      https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git
kernelci-pipeline-test_report |   SHA1:     9886142c7a2226439c1e3f7d9b69f9c7094c3ef6
kernelci-pipeline-test_report | --===============1230460839464074404==--
kernelci-pipeline-test_report | 
```

gctucker commented:
It would also be good to clean up the output in the logs by dropping the email headers. Only the report body should be printed, rather than the email. That could be done as a separate PR though.

JenySadadia commented:
Updated the script to subscribe for checkout node to be updated with 'completed' status.
Added code to get all child nodes from the root node and generate a test report.

gctucker commented on config/reports/test-report.jinja2:
```diff
@@ -1,7 +1,7 @@
 {# SPDX-License-Identifier: LGPL-2.1-or-later -#}
 
-{{ root.revision.tree }}/{{ root.revision.branch }} {{ '(Unknown)' if root.revision.describe is none else root.revision.describe}}: {{ total_runs }} runs, {{ total_failures}} failures
+{{ root.revision.tree }}/{{ root.revision.branch }} {{ '(Unknown)' if root.revision.describe is none else root.revision.describe}}: {{ total_runs }} runs, {{ total_failures}} failures, {{ total_timeouts }} timeouts
```
I don't think the number of timeouts is useful here, ideally only the number of regressions is needed in the email reports. Until we have that, we can just show the number of failures.

JenySadadia replied:
Done with the changes.

gctucker commented:
Adding skip label so we can test and merge #77

gctucker commented:
I've cherry-picked the last 2 commits of this PR into #90 so once it's merged I think this one can be closed.

gctucker commented:
The last 2 changes have been merged as part of #90 and the other ones are from #66 so this can be closed now.
